platform/mellanox/iproute2/scripts/get_sdk_patches.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/python3
+
+'''
+Script to download SDK iproute2 patches.
+
+If we build SONiC with SDK debians being taken from github, need to downlaod patches from github, usage will be:
+./get_sdk_patches.py -i <iproute2 version> -s <SDK version>
+
+If we build SONiC with SDK compiled from sources, need to take patches from SDK sources, usage will be:
+./get_sdk_patches.py -i <iproute2 version> -u <SDK sources url> 
+'''
+import argparse
+import requests
+import sys
+import os
+import subprocess
+from zipfile import ZipFile
+
+SDK_GITHUB_URL_BASE = 'https://raw.githubusercontent.com/Mellanox/Spectrum-SDK-Drivers/'
+SDK_GITHUB_URL_BASE = 'https://raw.githubusercontent.com/liorghub/Spectrum-SDK-Drivers/'
+SDK_IPROUTE2_PATCHES = '/iproute2/patches/'
+SDK_CODE_PACKAGE = 'code_package'
+SDK_CODE_PACKAGE_ZIP = SDK_CODE_PACKAGE + '.zip'
+PATCH_DIR = 'patch'
+SERIES_FILE = 'series'
+
+
+def download_file(base_url, file_name):
+    url = base_url + '/' + file_name
+    response = requests.get(url)
+    if response.status_code != requests.codes.ok:
+        raise Exception("Failed to download {}".format(url))
+    open(file_name, "wb").write(response.content)
+
+def check_proc(proc):
+    if proc.returncode:
+        raise Exception("Failed command: {}, stdout: {} stderr: {}".
+                        format(proc.args, proc.stdout.decode(), proc.stderr.decode()))
+
+def main():
+    rc = 0
+    try:
+        parser = argparse.ArgumentParser(description="Download SDK iproute2 patches")
+        parser.add_argument("-i", "--iproute2_version", required=True, help="iproute2 version")
+        parser.add_argument("-s", "--sdk_version", required=False, help="SDK version")
+        parser.add_argument("-u", "--sdk_url", required=False, help="SDK sources url")
+
+        args = parser.parse_args()
+        if (args.sdk_url is None and args.sdk_version is None):
+            raise Exception("Must supply one of the following arguments, sdk_version or sdk_url")
+
+        iproute2_series = open(PATCH_DIR + '/' + SERIES_FILE, "a")
+
+        # If we build SONiC with SDK debians taken from github, downlaod patches from github
+        if args.sdk_url is None:
+            # Download SDK series file
+            url = SDK_GITHUB_URL_BASE + args.sdk_version + SDK_IPROUTE2_PATCHES + args.iproute2_version
+            download_file(url, SERIES_FILE)
+
+            # Download patches to patch directory
+            with open(SERIES_FILE) as sdk_series_file:
+                for line in sdk_series_file:
+                    line = line.strip()
+                    if line.endswith('.patch') == False:
+                        continue
+                    # Download patch
+                    download_file(url, line.strip())
```

Comment:
Instead of making multiple requests, i think it's better to download the repo altogether as a zip file and act on it. Now that the bins are separated from sources, i think it's okay to download the folder.

Reply:
I have removed the script completely, now all operations are being performed from Makefile.